synapse/http/matrixfederationclient.py

```diff
@@ -164,7 +165,8 @@ async def _handle_json_response(
     try:
         check_content_type_is_json(response.headers)
 
-        d = treq.json_content(response)
+        # Reject Python extensions to JSON.
+        d = treq.json_content(response, parse_constant=_reject_invalid_json)
```

Comment:
Looks like this requires upping the minimum version of treq to 18.6.0. Do we have any issues with that?

Reply:
Fedora 31 (EOL Nov. 17) only has treq 17.8.0, so I decided to essentially [inline part of `json_content`](https://github.com/twisted/treq/blob/dcd63574b574595b30beb7b442329e995181f6a3/src/treq/content.py#L99-L101).